Tests/BackendIntegrationTests/StoreKitIntegrationTests.swift

```diff
@@ -421,6 +421,39 @@ class StoreKit1IntegrationTests: BaseStoreKitIntegrationTests {
         try await subscribe()
     }
 
+    func testSubscribeAfterExpirationWhileAppIsClosed() async throws {
+        // 1. Subscribe
+        let customerInfo = try await self.purchaseMonthlyOffering().customerInfo
+        let entitlement = try XCTUnwrap(customerInfo.entitlements[Self.entitlementIdentifier])
+
+        // 2. Simulate closing app
+        Purchases.clearSingleton()
+
+        // 3. Force several renewals while app is closed.
+        for _ in 0..<3 {
+            try self.testSession.forceRenewalOfSubscription(productIdentifier: entitlement.productIdentifier)
+        }
+
+        // 4. Expire subscription
+        try await self.expireSubscription(entitlement)
+
+        // 5. Re-open app
+        await self.resetSingleton()
+        self.logger.clearMessages()
+
+        // 6. Purchase again
+        _ = try await self.purchases.purchase(package: self.monthlyPackage)
```

Comment:
❗  One thing I don't understand even after looking at the receipt posted: this call to purchase returns an inactive entitlement ❓ 
We should look into it either in this PR or in a separate test.

Reply:
I'll wait to talk to @aboedo about this cause I'm really confused.

Reply:
yeah, that's unexpected - that's what this whole PR is meant to address. 
well, actually... 
I mean, purchase is going to return right after seeing one of the `queue` values return. like, this case is still true: 

<img width="919" alt="image" src="https://github.com/RevenueCat/purchases-ios/assets/3922667/dcba38d6-a048-47cf-9ff2-568c65024667">

The problem isn't just that the `initiation_source` was marked incorrectly, it's that we're returning that value for the purchase, when in reality we should ignore transactions that don't have that date entirely. 

I'll comment in the corresponding line